### PR TITLE
[Tech Debt] Remove stale api server code and add switch for dev mode - #98

### LINKED ISSUE #98: 🦺[Tech Debt] Derv
## Goal 
**Describe the work to be done**
 
There are some references to servers that are retired.  We also want to be able to test a backup server

## Definition of Done
- [ ] Refactor stale references
- [ ] Add dev server mode

### PR DESCRIPTION
**Pull request by vsima**
- fixes #98

### CODE REVIEW

**Comment by vsima** [comment]
@kcw-grunt i found another url that needs to be updated. hold off on this PR review for now.

File: app/build.gradle
Comment (kcw-grunt) [inline app/build.gradle:73]: thanks @vsima .  We have all the constants in BRConstants. It makes it easy to find.  Is there benefit from changing this?

File: app/src/main/java/com/breadwallet/presenter/entities/BRSettingsItem.java
Comment (kcw-grunt) [inline app/src/main/java/com/breadwallet/presenter/entities/BRSettingsItem.java:16]: This is a really good start. We need to make sure it's visible.  Will you add this:
- Status label
- Switch background color

| Settings Dev Mode: On | Settings Dev Mode: Off |
|--------------------|-------------------|
|Developer Mode_ Switch|Developer Mode_ Switch Off|

**Review by kcw-grunt — CHANGES_REQUESTED** [review]
We are really close. When I tested it I realized there is no direct way the non-tech users will know which mode they are in.  
It needs to be a big flag for them and our support guys to aid in troubleshotting.

This label should disappear when they are in prod mode and when prod server is down, the client should auto-switch to dev and all labels are updated.

Will you add this label in the Nav bar?:

|Dev Mode Label|
|----------------|
|Main with Dev mode on|

File: app/src/main/java/com/breadwallet/presenter/entities/BRSettingsItem.java
Comment (vsima) [reply app/src/main/java/com/breadwallet/presenter/entities/BRSettingsItem.java:16]: How does this look?

**Comment by vsima** [comment]
Dev mode copy added to the main screen:

File: app/src/main/java/com/breadwallet/presenter/entities/BRSettingsItem.java
Comment (kcw-grunt) [reply app/src/main/java/com/breadwallet/presenter/entities/BRSettingsItem.java:16]: ✅

**Review by kcw-grunt — APPROVED** [review]

**Comment by kcw-grunt** [comment]
cc: @vsima  the servers are not quite ready so I approved but not merging until after the 
#82 is all ready and we get a prod release out.

**Review by josikie — APPROVED** [review]
Great! @kcw-grunt @vsima 
Thanks!

I have some questions:
1. what is this syntax called: v -> {....} (AdvancedActivity.java, lines 141)?
2. what is an inflater?
3. what is android:foreground for, and why does it value selectableItemBackground?

**Comment by kcw-grunt** [comment]
will not do

**Pull request closed without merging**
